artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -1018,6 +1018,7 @@ private void sendContinuations(final int packetSize,
          currentLargeMessage.addBytes(body);
 
          if (!continues) {
+            currentLargeMessage.sync();
             currentLargeMessage.releaseResources();
```

Comment:
wouldn't releaseResources make a sync when the file is closed?

Reply:
Some other code is calling releaseResources such as LargeServerMessageImpl::deleteFile, LargeMessageDeliverer::finish which would cost an extra system call. We'd better sync when the large message file is first created and finished.
I just found ServerSessionImpl::messageToLargeMessage needs a sync as well.

Could we add a sync parameter in releaseResources? If true, file will be synced before closing.